src/celeritas/optical/interactor/WavelengthShiftInteractor.hh

```diff
@@ -0,0 +1,164 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/interactor/WavelengthShiftInteractor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/data/StackAllocator.hh"
+#include "celeritas/Types.hh"
+#include "celeritas/grid/GenericCalculator.hh"
+#include "celeritas/optical/Interaction.hh"
+#include "celeritas/optical/ParticleTrackView.hh"
+#include "celeritas/optical/TrackInitializer.hh"
+#include "celeritas/optical/WavelengthShiftData.hh"
+#include "celeritas/phys/InteractionUtils.hh"
+#include "celeritas/random/distribution/ExponentialDistribution.hh"
+#include "celeritas/random/distribution/IsotropicDistribution.hh"
+#include "celeritas/random/distribution/PoissonDistribution.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Apply the wavelength shift (WLS) to optical photons.
+ *
```

Comment:
Can you add a few sentences about the physical process of WLS and the underlying sampling methodology?

Reply:
No problem.  A short description of WLS is added.